handlers/discount-api/test/eligibilityChecker.test.ts

```diff
@@ -61,3 +64,22 @@ test('error', () => {
 		);
 	}).toThrow('Amount payable for next invoice');
 });
+
+test('Eligibility check works for supporter plus with 2 rate plans', () => {
+	const sub = zuoraSubscriptionSchema.parse(subSupporterPlusFullPrice);
+	const catalog = new ZuoraCatalogHelper(
+		zuoraCatalogSchema.parse(catalogJsonCode),
+	);
+	const billingPreview = billingPreviewSchema.parse(
+		billingPreviewSupporterPlusFullPrice,
+	);
+	const eligibilityChecker = new EligibilityChecker(catalog);
+	expect(
+		dayjs(
+			eligibilityChecker.getNextBillingDateIfEligible(
+				billingPreview,
+				getDiscountableRatePlan(sub),
+			),
+		).format('YYYY-MM-DD'),
+	).toEqual('2024-07-04');
```

Comment:
This is totally subjective so feel free to ignore! I'm a fan of the [arrange-act-assert](https://wiki.c2.com/?ArrangeActAssert) pattern in tests, just to make clear what's happening at each phase. What do you think about this slight refactoring?

```suggestion
        // Arrange
        // ..other setup
	const eligibilityChecker = new EligibilityChecker(catalog);
	
	// Act
	const nextDate = eligibilityChecker.getNextBillingDateIfEligible(
	    billingPreview,
	    getDiscountableRatePlan(sub),
	);
	
	// Assert
	expect(dayjs(nextDate).format('YYYY-MM-DD')).toEqual('2024-07-04');
```

Reply:
yep I'm totally on board with that, I never heard that way of describing it but I do try to follow it, it was a just a bit copy and paste at the moment.
I will check the other tests and keep them all consistent with your suggestion.

Reply:
although that's an interesting one for the tests that are expected to throw - because catching an exception is not referentially transparent, it's not easy to pull out the "Act" and the "Assert".
`Try` is a very scala-y thing so I'm guessing that or similar is not the done thing in TS?

Reply:
> although that's an interesting one for the tests that are expected to throw - because catching an exception is not referentially transparent, it's not easy to pull out the "Act" and the "Assert".

Yeah totally, I definitely find it harder to follow AAA in these cases.

> Try is a very scala-y thing so I'm guessing that or similar is not the done thing in TS?

True, it's not a language feature, but you do occasionally see abstractions which attempt to provide something similar, it's not very common.